### DISCUSSION TITLE
Version 1.2.0 Beta

### DISCUSSION THREAD

**Discussion by ABaumher** [ABaumher/galaxy-integration-steam#40] [Announcements]
This is a soft release of the changes we have been making in the background which should improve overall performance. It will not handle edge case errors for individual games and the like, but it should make the import process a lot faster. For users with large libraries, this should dramatically increase the capabilities of the plugin. 

Please note that this is a large departure from the old code. It may not be as bulletproof as the old code, but then again the old code is not very resilient to begin with. Please report any issues you find.

<hr /><em>This discussion was created from the release <a href='https://github.com/ABaumher/galaxy-integration-steam/releases/tag/'>Version 1.2.0 Beta</a>.</em>

**Comment by ABaumher** [comment]
@urwrstkn8mare can you get me a MacOS Release? This is based off the latest BetterProtoDeps on my repo. It includes your latest fixes.

**Comment by engiefox** [comment]
Stalled a little more than halfway through syncing games. No achieves/playtime. Worked just fine when I started it with an existing cache.
Had this trace in the log:
`2023-07-01 22:17:15,353 - steam_network.websocket_client - ERROR - Failed to establish authenticated WebSocket connection AssertionError()
2023-07-01 22:17:15,354 - steam_network.websocket_client - ERROR - Traceback (most recent call last):
  File "C:\Users\obliv\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\steam_network\websocket_client.py", line 113, in run
    await run_task
  File "C:\Users\obliv\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\steam_network\protocol_client.py", line 114, in run
    await self._protobuf_client.run()
  File "C:\Users\obliv\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\steam_network\protocol\protobuf_client.py", line 176, in run
    await self._process_packet(packet)
  File "C:\Users\obliv\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\steam_network\protocol\protobuf_client.py", line 498, in _process_packet
    await self._process_message(emsg, header, packet[8 + header_len:])
  File "C:\Users\obliv\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\steam_network\protocol\protobuf_client.py", line 509, in _process_message
    await self._process_client_logged_off(body)
  File "C:\Users\obliv\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\steam_network\protocol\protobuf_client.py", line 559, in _process_client_logged_off
    assert self._heartbeat_task is not None
AssertionError`
[GalaxyClient.log](https://github.com/ABaumher/galaxy-integration-steam/files/11928733/GalaxyClient.log)
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/11928734/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

**Reply by urwrstkn8mare** [reply]
```python
    async def _process_client_logged_off(self, body):
        logger.debug("Processing message ClientLoggedOff")
        message = CMsgClientLoggedOff()
        message.ParseFromString(body)
        result = message.eresult

        assert self._heartbeat_task is not None
        self._heartbeat_task.cancel()

        if self.log_off_handler is not None:
            await self.log_off_handler(result)
```
this seems to be the offending code incase anyone was curious

im not familiar with this part of the code so sry im of no help

**Reply by ABaumher** [reply]
Idk why that's an assert, a null state there is valid if not logged in yet. I don't believe it's always been that way but I'll double check

**Reply by ABaumher** [reply]
That said, at that point heartbeat should not be null, which likely means it was never stated, which would in turn be why we're getting that error. Time to track that down

It never ends.

**Comment by engiefox** [comment]
These keep showing in the log:
`2023-07-03 21:45:09,873 - steam_network.protocol.protobuf_client - WARNING - app 'Sid Meier's Civilization V SDK' has unexpected type 'tool'
2023-07-03 21:45:09,884 - steam_network.protocol.protobuf_client - WARNING - app 'Half-Life SDK' has unexpected type 'tool'
2023-07-03 21:45:09,889 - steam_network.protocol.protobuf_client - WARNING - app 'SCP: Secret Laboratory ServerMod' has unexpected type 'tool'
2023-07-03 21:45:09,894 - steam_network.protocol.protobuf_client - WARNING - app 'Left 4 Dead Authoring Tools' has unexpected type 'tool'
2023-07-03 21:45:09,896 - steam_network.protocol.protobuf_client - WARNING - Unrecognized app structure in app 520: KeyError('common')`
I don't know if stuff like this needs to be shown in GOG or not

edit: this probably should though
`2023-07-03 21:45:12,751 - steam_network.protocol.protobuf_client - WARNING - app 'OpenVR Benchmark' has unexpected type 'application'`

**Reply by engiefox** [reply]
Not sure how many people would want to launch their software through Steam going through GOG, but something to think about maybe?

**Reply by ABaumher** [reply]
An "app" is essentially anything steam offers, which includes things like videos, toolkits, etc. So naturally we get those alongside games and dlc. Those warnings are expected, they're telling us what we processed wasn't a game or dlc. Could drop severity to info but I'm leaving it for now. That 520 one is weird and I've seen it before. Needs looking into but I'm really prioritizing the big fixes over one or two apps acting up. 

As for launching toolkits or sdks through gog, that's not possible at this time. That'd have to be something gog supports and i don't think they will (ngl i wouldn't, either, as that's not really the focus of Galaxy client, which is just for games). We could probably fake it by treating them as games but that'd be more trouble than it's worth imo. We'd have to mark them as not having achievements so we don't waste cycles sending those requests, and making them adds complexity. There's other issues that'd undoubtedly pop up too and my brain already hurts enough as it is

**Comment by Mr-Spock1** [comment]
The auto-copy-mac.command has to have LF line termination, not CRLF. At least on Intel Macs. On Apple Silicon that might be different, I don't know.
Import works flawless. Even the very few Games, that refused to show up as installed in previous plugin version do so now.

**Reply by Mr-Spock1** [reply]
There is one problem with the game "Devour" ("The Watchers: Devour" in Galaxy):
The tile is empty. Normally I can replace the 3 Graphics in the "MEDIA" tab of the Edit dialogue, but the rectangle just shows the text "Could not load the Image". It won't accept a graphics file to drop over. The other two graphics are there.
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/12053070/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

**Comment by engiefox** [comment]
Crash after running for several hours, didn't give a trace, it might be the same crash from previous versions I've been getting.
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/12053399/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)
[GalaxyClient.log](https://github.com/ABaumher/galaxy-integration-steam/files/12053397/GalaxyClient.log)

**Comment by ninjagod888** [comment]
Hi, I've tried the installation procedure many times, but it always gave an endless "importing steam..." I did managed to import some games like half of them, but I don't even know how, and after I closed and open again the plugin said "offline retry" and after I logged with steam again even those little games were gone. I'm trying for hours to get this to work, please someone help me. PS: maybe it might cause problem that i have 6000+ games on steam, I have no idea what could cause the problem at this point

**Reply by ABaumher** [reply]
Please share your logs found at `%programdata%\GOG.com\Galaxy\logs`
We'll want to see the one for steam and the galaxy client. If you have multiple thousand games, you will have many different log files for steam, We'll need to see all of them. I recommend deletering those log files while gog of not running, then starting it up again so we can get clean log files. 

I'm willing to guess it's not able to get all the games in your library before gog thinks we've crashed (the plugin has gone too long without responding), at which point they shut us down. I've been working on a fix but it's slow going

**Reply by ninjagod888** [reply]
I have these 6 files I hope they can help
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.5.log](https://github.com/ABaumher/galaxy-integration-steam/files/13402663/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.5.log)
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/13402665/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.3.log](https://github.com/ABaumher/galaxy-integration-steam/files/13402661/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.3.log)
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.1.log](https://github.com/ABaumher/galaxy-integration-steam/files/13402658/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.1.log)
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.4.log](https://github.com/ABaumher/galaxy-integration-steam/files/13402662/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.4.log)

[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.2.log](https://github.com/ABaumher/galaxy-integration-steam/files/13402659/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.2.log)

**Comment by NovaStorm775** [comment]
sorry I'm a noob, but this plugin is dope, you guys are wizards! Let me know if I can send any logs through to help development! I just set up the plugin and its working like a charm!

**Comment by tinysapien** [comment]
I have never felt so stupid in my life, I cannot get this to work. I have deleted anything steam in the GOG folder, I tried completely wiping ALL GOG (as in proper uninstall) even game files to make sure it was just gone-gone, redownloaded both the fix and GOG, restarted my computer. I cannot get it to work, I'm still just getting 'Offline. Retry.'.

**Reply by ABaumher** [reply]
Mac OS or windows?

**Reply by tinysapien** [reply]
Windows - fully updated.

**Reply by engiefox** [reply]
Have you followed the instructions on the front page? The download files do not go with your game installation files, but in your AppData folder. That would look like: %localappdata%\GOG.com\Galaxy\plugins\installed

https://github.com/ABaumher/galaxy-integration-steam#installation

**Reply by tinysapien** [reply]
Yes, I am aware they don't go there - I was being cautious to try everything because I only have 2 GOG games and no saved files to worry about. I wouldn't have posted if I hadn't tried the literal instructions in the README.

**Reply by engiefox** [reply]
I'm just being thorough. Have you tried the 1.0.7 release instead of 1.2.0?

**Reply by tinysapien** [reply]
No change with 1.0.7. Even tried freshly making the folder in case of any wonky nonsense.

**Reply by ABaumher** [reply]
Delete your logs (location is on the repo front page) then run the application again. Then, attach those logs here. I'll take a look when back where there's internet

**Reply by tinysapien** [reply]
4th times a charm, apparently, or deleting all the logs mysteriously did something because now its connecting and I'm not going to even pretend to understand why - can still send the log from the successful one but since that won't have any of the previous debugs probably not worth it.

**Reply by ABaumher** [reply]
Deleting the cache if you did that may have done the trick. Let us know if it stops working.

**Comment by Morentius** [comment]
Hello,

sadly for some reason the plugin stopped working for me and I can't get it to work again, regaradless how many times or how many different versions I install.
 
I've tried the 1.2.0 Beta and also the 1.0.7 Version.

What am I doing wrong or what is the issue? 

I've attached my log below:
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/ABaumher/galaxy-integration-steam/files/13763017/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

**Reply by ABaumher** [reply]
Same error as someone else had on the old repo - yourcredentials expired. I'll put up an issue on this repo, and try to get an 1.0.8 and 1.2.1 release. In the meantime, try logging out and logging in again. You can do this in the plugin menu, simply disconnect and reconnect it

**Reply by Morentius** [reply]
Thanks, that did it!

**Comment by Rockingconnor** [comment]
now that I have downloaded the windows download what do i do next. I was told to to find a read.me file to learn how to get this working but i cant find one.
 p.s don't know anything about coding.
